models/Import/Service/ResultImporter.php

```diff
@@ -0,0 +1,369 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoResultServer\models\Import\Service;
+
+use common_exception_Error;
+use core_kernel_persistence_Exception;
+use oat\generis\model\data\Ontology;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoOutcomeRds\model\AbstractRdsResultStorage;
+use oat\taoResultServer\models\classes\ResultServerService;
+use oat\taoResultServer\models\Import\Exception\ImportResultException;
+use oat\taoResultServer\models\Import\Input\ImportResultInput;
+use taoResultServer_models_classes_ResponseVariable;
+use taoResultServer_models_classes_Variable;
+use Throwable;
+
+class ResultImporter
+{
+    private Ontology $ontology;
+    private ResultServerService $resultServerService;
+
+    public function __construct(Ontology $ontology, ResultServerService $resultServerService)
+    {
+        $this->ontology = $ontology;
+        $this->resultServerService = $resultServerService;
+    }
+
+    /**
+     * @param ImportResultInput $input
+     * @return void
+     * @throws core_kernel_persistence_Exception|Throwable|common_exception_Error|ImportResultException
+     */
+    public function importByResultInput(ImportResultInput $input): void
+    {
+        $resultStorage = $this->getResultStorage();
+        $deliveryExecutionUri = $input->getDeliveryExecutionId();
+        $testUri = $this->getTestUri($resultStorage, $deliveryExecutionUri);
+        $testScoreVariables = $this->getTestScoreVariables($resultStorage, $deliveryExecutionUri);
+
+        $resultStorage->getPersistence()->transactional(
+            function () use ($resultStorage, $input, $testUri, $deliveryExecutionUri, $testScoreVariables) {
```

Comment:
Maybe I'm wrong, but we can declare a return type for this.
But it's better to check and not critical 🙂
```php
function () use ($resultStorage, $input, $testUri, $deliveryExecutionUri, $testScoreVariables): void {
```